xds/internal/xdsclient/tests/resource_update_test.go

```diff
@@ -802,7 +802,7 @@ func (s) TestHandleClusterResponseFromManagementServer(t *testing.T) {
 			}
 			cmpOpts := []cmp.Option{
 				cmpopts.EquateEmpty(),
-				cmpopts.IgnoreFields(xdsresource.ClusterUpdate{}, "Raw"),
+				cmpopts.IgnoreFields(xdsresource.ClusterUpdate{}, "Raw", "LBPolicyJSON"),
```

Comment:
Is it not possible (or valuable) to check the `LBPolicyJSON` field?

What do you think about having an `Equal()` method on the `ClusterUpdate` type, where we can implement custom equality logic instead of depending on `cmpOpts`.

Reply:
The non determinism of the raw JSON bytes (i.e. whitespaces) is the reason I unmarshaled into balancer config instead of checking this field.
Your second point I wanted to do, but the problem is it doesn't pick up the cmp opts set that I scaled up on an Equal() method unfortunately
(I.e. other tests already had custom cmp opts so scale those up instead of Equal() which ignores those options).

Reply:
What do you mean by `scale up` here?

Reply:
Added to. I.e. other tests set cmpopts before checking Equality, which I added to with my new field ignored. However, on an Equal() method, it does not take into account the cmp opts already set by the test, making the tests fail. If there was some way of plumbing the set options into my own defined Equal() method, I would have done that, but I don't think there is. Thoughts? I wanted to do Equal to not have to add this option to every call site that checks equality, but couldn't see a way around it.

Reply:
I think what I'm asserting is that when I plumb stuff through Equal, other cmpopts (i.e. ignoring other fields) don't get taken into account, which breaks them.